R/countrycode_factory.R

```diff
@@ -0,0 +1,27 @@
+#' create new `countrycode` functions with alternative dictionaries and
+#' defaults
+#' 
+#' @export
+countrycode_factory <- function(
+  origin = NULL,
+  destination = NULL,
+  warn = NULL,
+  nomatch = NULL,
+  custom_dict = NULL,
+  custom_match = NULL) {
+
+  # default function
+  out = countrycode
+
+  # change defaults
+  if(!is.null(origin)) formals(out)$origin <- origin
+  if(!is.null(destination)) formals(out)$destination <- destination
+  if(!is.null(warn)) formals(out)$warn <- warn
+  if(!is.null(nomatch)) formals(out)$nomatch <- nomatch
+  if(!is.null(custom_dict)) formals(out)$custom_dict <- custom_dict
+  if (!is.null(custom_match)) formals(out)$custom_match <- custom_match
```

Comment:
You could make this generic:

```r
args <- formals()

for (name in names(args)) {
  if(!is.null(args[[name]])) formals(out)[[name]] <- args[[name]])
}
```

Reply:
Good idea, thanks!

Reply:
turns out that tests fail when I do this :(  Will try to think of another more generic approach eventually

Reply:
Sorry, my fault! I should have tested the proposed code 🙈

I realized that `formals()` doesn't return the actual arguments but the _formal_ ones of the function definition.

This actually works:

```r
for (name in names(formals())) {
    arg <- get(name)
    if (!is.null(arg)) {
        formals(out)[[name]] <- arg
    }
}
```

Reply:
looks like this works, thanks!